tests/README.md

```diff
@@ -0,0 +1,285 @@
+```
+
+#### Running a specific test
+
+To run a specific test, use the `-t` flag:
+
+```bash
+gulp e2e -t <test-name>
+# For example, to run the test for submitting a Flink statement with a SELECT query:
+gulp e2e -t "should submit Flink statement - SELECT"
+```
+
+#### Running the tests using a specific VS Code version
+
+If you wanted to run the tests in a specific VS Code version, you can set the `VSCODE_VERSION`
+environment variable:
+
+```bash
+    VSCODE_VERSION=1.93.2 gulp e2e
+```
+
+### Debugging the E2E tests
+
+Please refer to the [Playwright - Debugging Tests](https://playwright.dev/docs/debug) documentation
+for a detailed guide on debugging Playwright tests.
+
+To debug a particular test, set the `PWDEBUG` environment variable to `1` and run the test:
+
+```bash
+PWDEBUG=1 gulp e2e -t <test-name>
+```
+
+This will launch the test in debug mode, and you can use the Playwright inspector to debug the test.
+The Playwright inspector is extremely useful to pick locators in the VS Code window to use in the
+tests.
+
+### Writing E2E tests
+
+#### Getting familiar with the libraries we use
+
+1. [`vscode-test-playwright`](https://github.com/ruifigueira/vscode-test-playwright/) -- the main
+   library we use to run the Playwright tests in a VS Code Electron instance.
+
+   Here are some of the key features that the library provides (taken from its README):
+
+   - Enables direct interaction with VSCode UI elements using Playwright selectors.
+   - Allows programmatic interaction with VSCode APIs to simulate user actions or access internal
+     state.
+   - Captures detailed information about test execution, including screenshots, network requests,
+     and console logs.
+
+   See
+   [here](https://github.com/ruifigueira/vscode-test-playwright/blob/48b0eeb60c9e6bec3b77df032707155daffa8d74/src/index.ts#L24-L31)
+   for the list of test fixtures offered by the library.
+
+1. [`electron-playwright-helpers`](https://github.com/spaceagetv/electron-playwright-helpers?tab=readme-ov-file#functions)
+   -- like the name suggests, it offers useful helpers to stub the Electron dialog windows among
+   other utilities.
+
+Grep for `stubMultipleDialogs` to see how we use it to stub the Electron dialog windows during the
+Confluent Cloud authentication flow.
+
+#### Process for writing new tests
+
+Without trying to be too prescriptive, here's a process that has worked for us to efficiently write
+new E2E tests. Please update this section as we write more tests and learn about the best practices.
+
+You've been tasked with writing a new E2E test. You don't know where to start? I've got you covered.
+
+1. First, look at the existing tests in the `tests/e2e/specs` directory. If you find a test file
+   that covers similar functionality, you can use that as a reference. If not, go ahead and create a
+   new test file with the format `<feature-name>.spec.ts`.
+1. Alright, now -- here's a skeleton for the test file. You can use this as a starting point:
```

Comment:
I'd keep a more neutral/direct tone here while still highlighting that this instruction set is bound to change as we smooth out the E2E test dev process

Reply:
Thanks! That makes a lot of sense. I've made it more direct -- it's easier to parse too. Please a look and let me know if there's any other tweaks you'd like me to make.